features/steps/thens/attributes.py

```diff
@@ -143,3 +143,169 @@ def step_impl(context, inst):
     invalid_guid_chars = [char for char in inst if char not in valid_chars]
     if invalid_guid_chars:
         yield ValidationOutcome(inst=inst, expected="^[0-9A-Za-z_$]+$", observed=invalid_guid_chars, severity=OutcomeSeverity.ERROR)
+
+
+@gherkin_ifc.step('{attribute_compared} value must be {comparison_op} the expression: {expression}')
+def step_impl(context, inst, attribute_compared:str, comparison_op:str, expression:str):
+    '''
+    Compare a attribute to a expression based on attributes.
+
+    The {comparison_op} operator can be 'equal to', 'not equal to', 'greater than', 'less than', 'greater than or equal to', and 'less than or equal to'.
+
+    The {expression} should be composed by attribute values, and use the following operators:
+    + : addition;
+    - : subtraction;
+    * : multiplication;
+    / : division;
+    % : modulus;
+    ** : exponentiation.'''
+
+    operators = {
+        '+' : operator.add,
+        '-' : operator.sub,
+        '*' : operator.mul,
+        '/' : operator.truediv,
+        '%' : operator.mod,
+        '**' : operator.pow,
+        'equal to' : operator.eq,
+        'not equal to' : operator.ne,
+        'greater than' : operator.gt,
+        'less than' : operator.gt,
+        'greater than or equal to' : operator.ge,
+        'less than or equal to' : operator.le,
+    }
+
+    # Get compared attribute value
+
+    attr_compared_value = getattr(inst, attribute_compared, 'Compared attribute not found')
+    if isinstance(attr_compared_value, ifcopenshell.entity_instance):
+        raise Exception('Compared attribute value is an IFC entity')
+
+    # Replace attribute names with attribute values in the expression
+    for string in expression.split():
+        # Checks if the string is not a operator neither parenthesis
+        if string not in [*operators, '(', ')']:
+            if hasattr(inst, string):
+                if not isinstance(getattr(inst, string), ifcopenshell.entity_instance):
+                    expression = expression.replace(string, str(getattr(inst, string)))
+                else:
+                    raise Exception('Expression attribute value is an IFC entity')
+            else:
+                raise Exception('Expression attribute not found')
+
+    # Evaluate the string expression using eval
+    try:
+        expression_value = eval(expression)
+    except Exception as e:
+        raise ValueError(f"Error evaluating expression: {e}")
```

Comment:
I've asked this before but what's currently the behaviour of raising exceptions?

What happens for example with a `ZeroDivisionError`.

Regarding that specifically should there be some rules to prevent zeros in the denominator in this equation?

Reply:
> IIRC, in case there is an error unrelated to the rule (e.g. a ZeroDivisionError as mentioned earlier), it's a bit annoying for the gherkin CI/CD but without consequences for subsequent rules.

It would be nice to have something in the decorator or validation_handling that could return an ERROR outcome whenever an exception is raised.  I started to replace all of the `raise` statements with `yield ValidationOutcome(... ValidationSeverity.ERROR)` but that felt like a non-pythonic brute force approach.

Reply:
> What happens for example with a ZeroDivisionError.

It is not caught in CI/CD.  There is nothing in place to handle the exception and therefore the process fails before it can yield a ValidationOutcome with severity ERROR.  Current pytest logic is still `0 errors = pass`, not `all results are ValidationOutcome(ValidationSeverity.PASSED)`.

![image](https://github.com/user-attachments/assets/eaad6e1a-880f-4366-ae89-2d6b18b05c73)

Reply:
resolved in 6a3c82faa4f5f88871f93ece47fa74b9372cc53f